index/secondary_index_manager.cc

```diff
@@ -129,11 +129,13 @@ view_ptr secondary_index_manager::create_view_for_index(const index_metadata& im
         }
     }
 
-    for (auto& col : schema->clustering_key_columns()) {
-        if (col == *index_target) {
-            continue;
+    if (!index_target->is_static()) {
```

Comment:
This comment is for the commit message: the statement should be stronger. It's not only that we don't *need* columns from the base clustering key, we actually **rely** on the fact that they aren't there for our code to be correct. Thanks to not having clustering columns here, we avoid all the sad multiplication issues when these static column values would be changed.

Reply:
I reworded the message, hopefully it is better now.